### PR TITLE
chore(schema): update vite config

### PR DESCRIPTION
**Pull request by pi0**
### ❓ Type of change

- [ ] 📖 Documentation (updates to the documentation or readme)
- [ ] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

Add leftovers from #6220

### 📝 Checklist

- [ ] I have linked an issue or discussion.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: packages/schema/src/types/config.ts [hunk 1]
```diff
@@ -1,13 +1,14 @@
...
-import { Options as VuePluginOptions } from '@vitejs/plugin-vue'
+import type { UserConfig as ViteUserConfig } from 'vite'
+import type { Options as VuePluginOptions } from '@vitejs/plugin-vue'
 
 type DeepPartial<T> = T extends Function ? T : T extends Record<string, any> ? { [P in keyof T]?: DeepPartial<T[P]> } : T
 
 /** User configuration in `nuxt.config` file */
 export interface NuxtConfig extends DeepPartial<Omit<ConfigSchema, 'vite'>> {
-  vite?: ConfigSchema['vite']
+  // Avoid  DeepPartial for vite config inteface (https://github.com/nuxt/framework/pull/4772)
```
Comment (pi0) [inline packages/schema/src/types/config.ts:10]:
```suggestion
  // Avoid  DeepPartial for vite config interface (#4772)
```

File: packages/schema/src/types/config.ts [hunk 2]
```diff
@@ -1,13 +1,14 @@
...
-import { Options as VuePluginOptions } from '@vitejs/plugin-vue'
+import type { UserConfig as ViteUserConfig } from 'vite'
+import type { Options as VuePluginOptions } from '@vitejs/plugin-vue'
 
 type DeepPartial<T> = T extends Function ? T : T extends Record<string, any> ? { [P in keyof T]?: DeepPartial<T[P]> } : T
 
 /** User configuration in `nuxt.config` file */
 export interface NuxtConfig extends DeepPartial<Omit<ConfigSchema, 'vite'>> {
-  vite?: ConfigSchema['vite']
+  // Avoid  DeepPartial for vite config interface (#4772)
```
Comment (danielroe) [inline packages/schema/src/types/config.ts:10]:
```suggestion
  // Avoid DeepPartial for vite config interface (#4772)
```

File: packages/schema/src/types/config.ts [hunk 3]
```diff
@@ -1,13 +1,14 @@
...
-import { UserConfig } from 'vite'
-import { Options as VuePluginOptions } from '@vitejs/plugin-vue'
+import type { UserConfig as ViteUserConfig } from 'vite'
+import type { Options as VuePluginOptions } from '@vitejs/plugin-vue'
 
 type DeepPartial<T> = T extends Function ? T : T extends Record<string, any> ? { [P in keyof T]?: DeepPartial<T[P]> } : T
 
 /** User configuration in `nuxt.config` file */
 export interface NuxtConfig extends DeepPartial<Omit<ConfigSchema, 'vite'>> {
-  vite?: ConfigSchema['vite']
```
Comment (danielroe) [inline packages/schema/src/types/config.ts:10]: Why this change? Seems like this would override the schema types entirely.

Comment (pi0) [reply packages/schema/src/types/config.ts:10]: They are same reference now with new ViteConfig interface.

Comment (danielroe) [reply packages/schema/src/types/config.ts:10]: So why not keep it as `ConfigSchema['vite']`? It would be confusing for a future contributor to edit schema and not have it take effect.

File: packages/schema/src/types/config.ts [hunk 4]
```diff
@@ -1,13 +1,14 @@
...
+  // Avoid DeepPartial for vite config interface (#4772)
+  vite?: ViteConfig
```
Comment (danielroe) [inline packages/schema/src/types/config.ts:11]:
```suggestion
  vite?: ConfigSchema['vite']
```

File: packages/schema/src/types/config.ts (hunk 3 above)
Comment (pi0) [reply packages/schema/src/types/config.ts:10]: Actually, I was confused at first too what is the difference between these two and had to check. 

Also as I can understand, we do not infer this type from schema defaults. Two NuxtConfig and NuxtOptions refer to the same custom interface.

Comment (danielroe) [reply packages/schema/src/types/config.ts:10]: I think it is better to keep as ConfigSchema. Logically we are modifying it only to apply DeepPartial to _part_ of it. So both forks should keep reference to ConfigSchema, even if it ultimately resolves to ViteConfig. It is more consistent.
